### ISSUE TITLE
pip plugin: some aliases delete existent requirements.txt

### ISSUE THREAD

**Issue by ihuro** [ohmyzsh/ohmyzsh#9977]
If `requirements.txt` exists, `pipupall` and `pipunall` aliases update it and then remove it.
Would be better to use a different filename in this case.

https://github.com/ohmyzsh/ohmyzsh/blob/3e7998aec33ff7551d06d4510f9e8769aab5db93/plugins/pip/pip.plugin.zsh#L91
https://github.com/ohmyzsh/ohmyzsh/blob/3e7998aec33ff7551d06d4510f9e8769aab5db93/plugins/pip/pip.plugin.zsh#L97

**Comment by zcutlip** [comment]
I'd recommend having the alias create a temporary requirements file. A shell function may be easier, because the quote escaping gets complicated.

For demonstration, here's a start that generates a temporary requirements file, then echoes it's name and then cats it:

```sh
alias piptemp='tempreq=$(mktemp) && pip freeze > $tempreq && echo $tempreq && cat $tempreq
```

**Comment by zcutlip** [comment]
Additionally, these aliases probably need some additional refinement to deal with:

- packages installed in editable mode: `pip install -e`
- local packages installed `foo @ file:///path/to/package`
- packages installed from git URLs: `pip install git+https://github.com/.../foo.git`
- packages installed globally vs user: `--user`

These aliases are quite brute-force and I would not recommend using them without some refinement.

**Comment by mcornella** [comment]
Fixed in #10061.
